spec/models/message_spec.rb

```diff
@@ -0,0 +1,80 @@
+# == Schema Information
+#
+# Table name: messages
+#
+#  id              :bigint           not null, primary key
+#  text            :text             not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  conversation_id :bigint           not null
+#  user_id         :bigint
+#
+# Indexes
+#
+#  index_messages_on_conversation_id  (conversation_id)
+#  index_messages_on_user_id          (user_id)
+#
+
+require 'rails_helper'
+require 'support/shared_context/conversation_between_two_users.rb'
+
+describe Message, type: :model do
+  include_context 'conversation_between_two_users'
+
+  subject do
+    build(:message, user: user, conversation: conversation)
+  end
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:text) }
+    it { is_expected.to belong_to(:conversation) }
+  end
+
+  describe '#notify' do
+    context 'when message is target match' do
+      let!(:message) do
+        create(:message, conversation: conversation,
+                         user: user,
+                         text: 'You have a new match!')
+      end
+
+      it 'does not enqueue job' do
```

Comment:
I'm not sure what are you testing here, since the message creation is defined with a `!` (this makes the massage to be create before executing any test code)

Reply:
This tests than when the function notify on the message is called, if it was a target match message, it won't notify anyone.